### PR TITLE
capture custom role creation in prehog

### PR DESCRIPTION
**Pull request by michellescripts**
supports https://github.com/gravitational/cloud/issues/2521
prehog/cloud partner: https://github.com/gravitational/cloud/pull/3313/files

Captures an audit event for "create" role (which also runs for update)
Passes the UI events through to the authenticated user events web endpoint ("/webapi/capture")

Screenshot 2023-02-07 at 4 31 47 PM
Screenshot 2023-02-07 at 4 31 53 PM

### CODE REVIEW

File: proto/prehog/v1alpha/teleport.proto [hunk 1]
```diff
@@ -138,13 +138,6 @@ enum DiscoverResource {
   DISCOVER_RESOURCE_APPLICATION_HTTP = 8;
   DISCOVER_RESOURCE_APPLICATION_TCP = 9;
   DISCOVER_RESOURCE_WINDOWS_DESKTOP = 10;
-  DISCOVER_RESOURCE_DATABASE_SQLSERVER_RDS = 11;
-  DISCOVER_RESOURCE_DATABASE_POSTGRES_REDSHIFT = 12;
-  DISCOVER_RESOURCE_DATABASE_SQLSERVER_SELF_HOSTED = 13;
-  DISCOVER_RESOURCE_DATABASE_REDIS_SELF_HOSTED = 14;
-  DISCOVER_RESOURCE_DATABASE_POSTGRES_GCP = 15;
-  DISCOVER_RESOURCE_DATABASE_MYSQL_GCP = 16;
-  DISCOVER_RESOURCE_DATABASE_SQLSERVER_GCP = 17;
```
Comment (ryanclark) [inline proto/prehog/v1alpha/teleport.proto:147]: Should these be removed?

**Review by ryanclark — APPROVED** [review]

File: proto/prehog/v1alpha/teleport.proto (hunk 1 above)
Comment (michellescripts) [reply proto/prehog/v1alpha/teleport.proto]: Definitely not,  I reset the proto again!

**Review by espadolini — APPROVED** [review]
Should the preset roles be left in cleartext by the anonymization?

Should we also log role deletes, so we actually get to know if the recreation of a role is an edit or a delete-then-create-anew? Do we even care about the distinction?

**Comment by michellescripts** [comment]
@espadolini added cleartext default roles [here](https://github.com/gravitational/teleport/pull/21123/commits/235467efac1d891bb83b1ad46b561ed6509dc2a1).

I'll open a new PR for delete if we decide to add that through.

File: lib/services/usagereporter.go [hunk 2]
```diff
@@ -68,6 +69,8 @@ const (
 	usageReporterRetryAttempts = 5
 )
 
+var defaultTeleportRoles = []string{"access", "auditor", "editor"}
```
Comment (espadolini) [inline lib/services/usagereporter.go:72]: Move this to `constants.go`, under `PresetAuditorRoleName` (and use those consts).

Might wanna add `admin` to the list, too.

Comment (michellescripts) [reply lib/services/usagereporter.go]: Ouu, [done](https://github.com/gravitational/teleport/pull/21123/commits/d0703124270a858fd173958ebbf95683ee7d36e6)

**Review by espadolini — APPROVED** [review]

**Pull request merged**